keras/src/ops/operation.py

```diff
@@ -23,7 +24,11 @@ def __init__(self, dtype=None, name=None):
                 "cannot contain character `/`. "
                 f"Received: name={name} (of type {type(name)})"
             )
-        self._dtype_policy = dtype_policies.get(dtype)
+        if isinstance(dtype, dtype_policies.DTypePolicyMap):
+            self._dtype_policy = dtype_policies.get(dtype[name])
```

Comment:
I think I am still missing something here. How is this supported to work for many layers of nesting? E.g. the KerasNLP case, where Model -> Block x 12 -> MHA x 2 -> Dense x 4.

In some cases, child layers are only created on `build()`, how do you proxy this object all the way down in that case? What the new best practice way to write a composite layer (potentially of more composite layers)?

And is this line still valid https://github.com/keras-team/keras/blob/v3.3.3/keras/src/layers/attention/multi_head_attention.py#L291